transifex/native/tools/migrations/mark.py

```diff
@@ -0,0 +1,159 @@
+MARK_PROOFREAD_FILE = 'Transifex Native: PROOFREAD_FILE'
+
+
+def mark_string(string_migration, comment_format, mark):
+    """Mark a string migration for proofreading.
+
+    Adds a comment in the beginning of this string.
+
+    :param StringMigration string_migration: the string migration object
+        to update
+    :param unicode comment_format: the format of the comment,
+        compatible with the file type of the migration
+    :param str mark: the string to add as a comment
+    """
+    string_migration.update(
+        '', comment_format.format(mark),
+        append=False,
+    )
+
+
+class MarkPolicy(object):
+    """Determines if a migrated file or a migrated string will be marked
+    for proofread.
+    """
+
+    def __init__(self):
+        # This determines how review comments will appear
+        # inside the migrated file. It is dynamic because different files
+        # support different comment formats, e.g.
+        # - '# Comment' in Python
+        # - '<!-- Comment -->' in HTML
+        self._comment_format = '{}'
+
+    def set_comment_format(self, comment_format):
+        """Define the comment format to use when a policy adds
+        comments to the migrated file.
+
+        Example:
+        >>> set_comment_format('# {}')
+
+        :param unicode comment_format: the format to use; must include '{}'
+        :raise ValueError: if `comment_format` does not include '{}'
+        """
+        if '{}' not in comment_format:
+            raise ValueError('The provided `comment_format` must include {}')
+        self._comment_format = comment_format
+
+    def should_mark_strings(self):
+        """Returns whether or not a policy object wants to mark individual
+        strings.
+
+        Although policy subclasses can only mark *some* strings, and not all,
+        this method is important for optimization, so that the objects that
+        use a policy object know if they need to feed each string migration
+        to the mark policy or not.
+        """
+        return False
+
+    def mark_file(self, file_migration):
+        """The base class does nothing.
+
+        :param FileMigration file_migration: the migration object
+        :return: True if the file was marked, False otherwise
+        :rtype: bool
+        """
+        return False
+
+    def mark_string(self, string_migration):
+        """The base class does nothing.
+
+        :param StringMigration string_migration: the migration object
+        :return: True if the string was marked, False otherwise
+        :rtype: bool
+        """
+        return False
+
+
+class NoopMarkPolicy(MarkPolicy):
+    """Does not mark anything."""
+
+    name = 'none'
+
+
+class MarkLowConfidenceFilesPolicy(MarkPolicy):
+    """Marks all files that have at least one string with low confidence."""
+
+    name = 'file-low'
+
+    def mark_file(self, file_migration):
+        if not len(file_migration.low_confidence_strings):
```

Comment:
`len() == 0`, make sense of boolean conditions. I think `not x with x being an integer` is a little bit abusive.

Reply:
Actually it should have been `if not file_migration.low_confidence_strings`, I somehow managed to mix the implicit boolean conversion with `len()`...